Apps/W1/EDocument/app/src/Processing/EDocumentSubscription.Codeunit.al

```diff
@@ -198,7 +198,36 @@ codeunit 6103 "E-Document Subscription"
     local procedure OnBeforeOnDeletePurchaseHeader(var PurchaseHeader: Record "Purchase Header"; var IsHandled: Boolean)
     begin
         if not IsNullGuid(PurchaseHeader."E-Document Link") then
-            Error(DeleteNotAllowedErr);
+            if GuiAllowed() then
+                if not Confirm(this.ConfirmDeleteQst) then
+                    Error('');
+    end;
+
+    [EventSubscriber(ObjectType::Table, Database::"Purchase Header", 'OnAfterDeleteEvent', '', false, false)]
+    local procedure OnAfterDeletePurchaseHeader(var Rec: Record "Purchase Header"; RunTrigger: Boolean)
+    var
+        EDocument: Record "E-Document";
+        EDocumentService: Record "E-Document Service";
+        EDocumentLog: Codeunit "E-Document Log";
+        EDocumentProcessing: Codeunit "E-Document Processing";
+        EDocServiceStatusDeleted: Enum "E-Document Service Status";
+    begin
+        if IsNullGuid(Rec."E-Document Link") then
+            exit;
+
+        EDocument.GetBySystemId(Rec."E-Document Link");
```

Comment:
If we cant find it, lets just exit. We should probably not block delete of header.